### PR TITLE
When Flask activation is missing, do not emit a log message

### PR DESCRIPTION
**Pull request by jpmelos**
# Description

If a Flask request doesn't have an active span, it just means that it was initialized via a mechanism that doesn't run `before_request`, like `app.test_request_context` or even manually. It is okay and instrumentation still works.

Fixes https://github.com/open-telemetry/opentelemetry-python-contrib/issues/160.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I ran the same script that is in https://github.com/open-telemetry/opentelemetry-python-contrib/issues/160 and I didn't get the warning message.

# Does This PR Require a Core Repo Change?

- [ ] Yes
- [x] No

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/master/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [ ] ~Unit tests have been added~ Not needed
- [ ] ~Documentation has been updated~ Not needed

### CODE REVIEW

**Review by lzchen — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#212](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/212))
 - `opentelemetry-instrumentation-fastapi` Added support for excluding some routes with env var `OTEL_PYTHON_FASTAPI_EXCLUDED_URLS`
   ([#237](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/237))
+- `opentelemetry-instrumentation-flask` Do not emit a warning message for request contexts created with `app.test_request_context` or manually.
```
Comment (codeboten) [inline CHANGELOG.md:24]: This should probably go under `Changed`

**Review by codeboten — APPROVED** [review]
Thanks for the PR! The branch needs updating and I added a comment regarding where in the changelog this should be. Once the branch is updated we can merge!

File: CHANGELOG.md (hunk 1 above)
Comment (jpmelos) [reply CHANGELOG.md]: Done in https://github.com/open-telemetry/opentelemetry-python-contrib/pull/253/commits/7a97ccdef2c9d86cf8bbd63d99a252dc915c944a.

**Comment by jpmelos** [comment]
@codeboten since my changes are a merge of `master` into this branch and an edit to a `.md` file, I have no idea how two checks failed when they had previously passed. Is there any chance those failures are intermittent and retrying would suffice? I can't find the retry button, so maybe I lack permission to retry actions?

**Comment by lzchen** [comment]
@jpmelos 
Can you try rebasing from master? The build should be fixed.

**Comment by jpmelos** [comment]
@lzchen all checks passed now.

**Pull request merged**